lib/emque/producing/message/message.rb

```diff
@@ -97,21 +99,36 @@ def to_json
         Oj.dump(data, :mode => :compat)
       end
 
-      def publish(publisher=Emque::Producing.publisher)
+      def publish(publisher=nil)
+        publisher = Emque::Producing.publisher unless publisher
         log "publishing...", true
         if valid?
           log "valid...", true
           if Emque::Producing.configuration.publish_messages
-            sent = publisher.publish(topic, message_type, to_json, partition_key)
-            log "sent #{sent}"
-            if raise_on_failure? && !sent
-              raise MessagesNotSentError.new
+            begin
+              sent = publisher.publish(topic, message_type, to_json, partition_key)
+              log "sent #{sent}"
+              if raise_on_failure? && !sent
+                raise MessagesNotSentError.new
+              end
+            rescue *PUBLISH_IGNORED_EXCEPTIONS => error
+              if raise_on_failure?
+                raise
+              else
+                log "failed ignoring exception... #{error}", true
+              end
             end
           end
         else
           log "failed...", true
           raise InvalidMessageError.new(invalid_message)
         end
+      rescue *PUBLISH_IGNORED_EXCEPTIONS => error
```

Comment:
How about making this configurable on the instance as well... The instance variable could inherit from the class's configuration by default, but allow for conditional toggling of raises at runtime.

Simple console exercise:
```ruby
@rescue_on = []

def test(error_class)
  raise error_class, "As requested :)"
rescue *@rescue_on
  puts "error happened and rescued"
end
 
test StandardError
# StandardError: As requested :)
# 	from (irb):5:in `test'
# 	from (irb):9
# 	from /Users/danmatthews/.rvm/rubies/ruby-2.1.2/bin/irb:11:in `<main>'

@rescue_on << StandardError

test StandardError
# error happened and rescued
# => nil 
```